### ISSUE TITLE
Using a public property extensions inside a class or object with a @Composable backing delegate fails to link for iOS targets

### ISSUE THREAD

**Issue by MikolajKakol** [JetBrains/compose-multiplatform#3084]
**Describe the bug**
Given code should compile on iOS as it does on Android, similar to #2901 

```
class G {
    val text: String @Composable get() = "Some text"
}
```
or 
```
object G {
    val text: String @Composable get() = "Some text"
}
```

It fails with
```
error: There is still an unbound symbol after generation of IR module <shared>:
Unbound public symbol IrSimpleFunctionPublicSymbolImpl: /G.text.<get-text>|8349765969627210272[0]
```

**Affected platforms**
Select one of the platforms below:
- iOS

**Versions**
- Kotlin version*: 1.8.20
- Compose Multiplatform version*: 1.4.0 and 1.5.0-dev1030
- OS version(s)* (required for Desktop and iOS issues): 16.2
- OS architecture (x86 or arm64): x86

**To Reproduce**
As in #2901

**Expected behavior**
It should compile

**Comment by pablichjenkov** [comment]
I can confirm this.

**Comment by TheMrDOC111** [comment]
I also have the same problem

**Comment by FireWookie** [comment]
i have this problem

**Comment by eymar** [comment]
There is a possible workaround described here: https://github.com/JetBrains/compose-multiplatform/issues/3175#issuecomment-1557252486

One more option is to change the visibility of the class if it's okay:

```
private class G {...}
internal class G {...}
```
___

It seems that in kotlin 1.9.0 it will be possible to add `@HiddenFromObjC` annotation to the class - https://youtrack.jetbrains.com/issue/KT-56464/K-N-Allow-HiddenFromObjC-for-classes
That annotation should fix this issue.
With kotlin 1.9.0, we'll patch the compiler plugin to add such an annotation automatically as we already do for functions. It's not possible to add this annotation in koltin 1.8.x to classes.

**Comment by eymar** [comment]
Hey everyone!
The bugs were fixed. Please update to 1.4.1 Compose Multiplatform, and use kotlin 1.8.21 with 

```kotlin
compose {
    kotlinCompilerPlugin.set("1.4.7")
}
```

In the next release, setting `kotlinCompilerPlugin.set("1.4.7")` won't be necessary.

**Comment by pablichjenkov** [comment]
Indeed, the problem seems to be gone with above configuration. Thanks

**Comment by MikolajKakol** [comment]
It doesn't quite work for me as
   > Compose Multiplatform 1.4.1 doesn't support Kotlin 1.8.21.

**Comment by MikolajKakol** [comment]
Yes, I added that in the shared module in the root and I saw that error message. 
<img width="1415" alt="obraz" src="https://github.com/JetBrains/compose-multiplatform/assets/773952/28bdfbb9-9b08-4003-b1f3-ab4c5d33eb0f">

**Comment by eymar** [comment]
Is "wed" also a module?
The error message comes from the compose gradle plugin. So it's likely there are modules with compose.

**Comment by pablichjenkov** [comment]
You need to add bellow code in all modules build.gradle file.
```
compose { 
     // Sets a specific JetBrains Compose Compiler version 
     kotlinCompilerPlugin.set("1.4.7") 
     kotlinCompilerPluginArgs.add("suppressKotlinVersionCompatibilityCheck=1.8.21") 
 }
```

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3084">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
